src/PrestaShopBundle/EventListener/Context/Admin/ShopContextListener.php

```diff
@@ -80,4 +81,28 @@ public function onKernelRequest(RequestEvent $event): void
             $this->shopContextBuilder->setShopId($shopConstraint->getShopId()->getValue());
         }
     }
+
+    private function getShopConstraintFromCookie(): ?ShopConstraint
```

Comment:
it would make this function easier to understand if we had a comment showing what the cookie value looks like before being parsed